### PR TITLE
Bug 1468351

### PR DESCRIPTION
**Pull request by fbertsch**

### CODE REVIEW

File: src/main/scala/com/mozilla/telemetry/views/Longitudinal.scala [hunk 1]
```diff
@@ -718,8 +723,14 @@ object LongitudinalView {
   }
   // scalastyle:on return
 
-  def getStandardHistogramSchema(schema: Schema): Schema = {
-    getElemType(schema, "fx_tab_switch_total_ms")
+  // ref: http://fdahms.com/2015/10/14/scala-and-the-transient-lazy-val-pattern/
+  // this val will only be calculated once per partition, and will not be serialized
```
Comment (sunahsuh) [inline src/main/scala/com/mozilla/telemetry/views/Longitudinal.scala]: Oh this is cool!

File: src/main/scala/com/mozilla/telemetry/views/Longitudinal.scala [hunk 2]
```diff
@@ -718,8 +723,14 @@ object LongitudinalView {
   }
   // scalastyle:on return
 
-  def getStandardHistogramSchema(schema: Schema): Schema = {
-    getElemType(schema, "fx_tab_switch_total_ms")
```
Comment (sunahsuh) [inline src/main/scala/com/mozilla/telemetry/views/Longitudinal.scala]: I think I need a bit more context on the fix -- on first glance, getStandardHistogramSchema and  standardHistogramSchema don't seem equivalent but it looks like it's being used as a drop-in replacement

Comment (sunahsuh) [reply src/main/scala/com/mozilla/telemetry/views/Longitudinal.scala]: Wow, that's convoluted :D I see that histogram stopped recording in 57, so interesting that this didn't fail earlier.

File: src/main/scala/com/mozilla/telemetry/views/Longitudinal.scala [hunk 3]
```diff
@@ -1036,7 +1047,8 @@ object LongitudinalView {
       Option(root.build)
     } catch {
       case e: Throwable =>
-        // Ignore buggy clients
+        // Log buggy clients errors and continue
+        logger.warn(org.apache.commons.lang.exception.ExceptionUtils.getStackTrace(e))
```
Comment (sunahsuh) [inline src/main/scala/com/mozilla/telemetry/views/Longitudinal.scala]: I have a tiny concern about logging the entire trace for all the bad clients blowing up our logs, but we can probably ignore that until it actually happens

File: src/main/scala/com/mozilla/telemetry/views/Longitudinal.scala [hunk 4]
```diff
@@ -730,14 +741,14 @@ object LongitudinalView {
                                    histogramDefinitions: Map[String, HistogramDefinition]) {
     val histogramsList = payloads.map(Histograms.stripKeyedHistograms)
 
-    val uniqueKeys = histogramsList.flatMap(x => x.keys).distinct.toSet
+    val uniqueKeys = histogramsList.flatMap(x => x.keys).distinct
```
Comment (sunahsuh) [inline src/main/scala/com/mozilla/telemetry/views/Longitudinal.scala]: Amazing what innocuous things cause this job to fail

File: src/main/scala/com/mozilla/telemetry/views/Longitudinal.scala (hunk 3 above)
Comment (fbertsch) [reply src/main/scala/com/mozilla/telemetry/views/Longitudinal.scala]: Good point - let's keep an eye on this.

**Pull request merged**